### PR TITLE
reuse record comparison code from QualityIO tests

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request moves the fastq converter test code from `test_SeqIO_converter.py` to `test_SeqIO_QualityIO.py` to be able to reuse the complicated record comparison code for quality score truncation.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by mdehoon** [comment]
Error seems unrelated

**Comment by peterjc** [comment]
TravisCI seems to have problems showing the failing log, rerunning the style check in case that helps.

File: Tests/test_SeqIO.py [hunk 1]
```diff
@@ -124,6 +124,45 @@ def compare_records(self, old_list, new_list, *args, **kwargs):
             self.compare_record(old, new, *args, **kwargs)
 
 
+class SeqIOConverterTestBaseClass(SeqIOTestBaseClass):
+
+    def check_conversion(self, filename, in_format, out_format, alphabet):
```
Comment (peterjc) [inline Tests/test_SeqIO.py:129]: Could you add a docstring (shown as the test name when run)?

File: Tests/test_SeqIO.py [hunk 2]
```diff
@@ -124,6 +124,45 @@ def compare_records(self, old_list, new_list, *args, **kwargs):
...
+        for record1, record2 in zip(records, records2):
+            self.compare_record(record1, record2, msg=msg)
+        # Finally, use the convert function, and check that agrees:
+        handle2 = StringIO()
+        with warnings.catch_warnings():
+            SeqIO.convert(filename, in_format, handle2, out_format, alphabet)
+        # We could re-parse this, but it is simpler and stricter:
+        self.assertEqual(handle.getvalue(), handle2.getvalue(), msg=msg)
+
+    def failure_check(self, filename, in_format, out_format, alphabet):
```
Comment (peterjc) [inline Tests/test_SeqIO.py:149]: Could you add a docstring (shown as the test name when run)

File: Tests/test_SeqIO_Insdc.py [hunk 3]
```diff
@@ -101,6 +104,22 @@ def test_annotation3(self):
         self.check_rewrite("EMBL/AE017046.embl")
 
 
+class ConvertTestsInsdc(SeqIOConverterTestBaseClass):
+
+    def test_conversion(self):
```
Comment (peterjc) [inline Tests/test_SeqIO_Insdc.py:109]: Could you add a docstring (shown as the test name when run)

**Comment by mdehoon** [comment]
@peterjc  Done, thanks.

**Comment by peterjc** [comment]
And TravisCI is happy now - a false alarm indeed.

**Pull request merged**